### PR TITLE
add test_543176_CreatingMultipleSnapshots

### PR DESCRIPTION
**Pull request by AlexandraAllakhverdieva**

### CODE REVIEW

File: mos_tests/cinder/cinder_tests.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                    image.name == 'TestVM'][0]
+        volume = common_functions.create_volume(self.cinder, image_id, 60)
+        self.volume_list.append(volume)
+
+        # 2. Creation of 70 snapshots
+        count = 70
+        initial_snapshot_lst = []
+        for num in xrange(count):
+            snapshot_id = self.cinder.volume_snapshots \
+                .create(volume.id, name='1st_creation_{0}'.format(num + 1))
```
Comment (TimurNurlygayanov) [inline mos_tests/cinder/cinder_tests.py]: why you use num + 1 here?

File: mos_tests/cinder/cinder_tests.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        # 3. Delete all snapshots
+        for snapshot in initial_snapshot_lst:
+            self.cinder.volume_snapshots.delete(snapshot)
+
+        # 4. Launch creation of 50 snapshot without waiting of deletion
+        new_count = 50
+        new_snapshot_lst = []
+        for num in xrange(new_count):
+            snapshot_id = self.cinder.volume_snapshots \
+                .create(volume.id, name='2nd_creation_{0}'.format(num + 1))
```
Comment (TimurNurlygayanov) [inline mos_tests/cinder/cinder_tests.py]: num + 1 ?

File: mos_tests/cinder/cinder_tests.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            new_snapshot_lst.append(snapshot_id)
+        self.snapshot_list.extend(new_snapshot_lst)
+
+        timeout = time.time() + (new_count + count) * 10
+        while [s.status for s in self.cinder.volume_snapshots.list() if
+               s in new_snapshot_lst].count('available') != new_count:
+            if time.time() < timeout:
+                time.sleep(10)
+            else:
+                raise AssertionError("At least one snapshot is not available")
```
Comment (TimurNurlygayanov) [inline mos_tests/cinder/cinder_tests.py]: it is not informative message, let's add information about the count of volumes here.

File: mos_tests/functions/common.py [hunk 4]
```diff
@@ -409,3 +420,37 @@ def check_volume_status(cinderclient, uid, status, timeout=5):
...
+            :param status: Expected volume snapshot status
+            :param timeout: Timeout for check operation
+            :return True or False
+    """
+    if check_volume_snapshot(cinderclient, uid):
+        start_time = time()
+        snapshot_status = [s.status
+                           for s in cinderclient.volume_snapshots.list()
+                           if s.id == uid.id][0]
+        while snapshot_status != status and time() < start_time + 60 * timeout:
```
Comment (TimurNurlygayanov) [inline mos_tests/functions/common.py]: while time() < start_time + 60 \* timeout:
____sleep(1)
____snapshot_status = [s.status
____________________for s in cinderclient.volume_snapshots.list()
____________________if s.id == uid.id][0]
____if snapshot_status == status:
________return True

Comment (gdyuldin) [inline mos_tests/functions/common.py]: Or just use https://pypi.python.org/pypi/waiting

File: mos_tests/cinder/cinder_tests.py (hunk 3 above)
Comment (AlexandraAllakhverdieva) [inline mos_tests/cinder/cinder_tests.py]: Do you mean count of unavailable snapshots?

Comment (TimurNurlygayanov) [inline mos_tests/cinder/cinder_tests.py]: yes.

**Pull request merged**